project/src/pages/main-page/main-page.tsx

```diff
@@ -2,11 +2,27 @@ import Footer from '../../components/common/footer/footer';
 import Header from '../../components/common/header/header';
 import PlayButton from '../../components/play-button/play-button';
 import FilmsList from '../../components/films-list/films-list';
-import {FilmsListProps} from '../../types/film';
-
+import {useAppSelector} from '../../hooks';
+import {selectFilterFilms} from '../../store/select';
+import {useDispatch} from 'react-redux';
+import GenreMenu from '../../components/genre-menu/genre-menu';
+import {useEffect, useState} from 'react';
+import {getFilmsList} from '../../store/action';
+import {Film, FilmsListProps} from '../../types/film';
+import ButtonShowMore from '../../components/button-show-more/button-show-more';
+import {FILMS_COUNT_PER_STEP} from '../../const';
 
 function MainPage({films}: FilmsListProps): JSX.Element {
   const {id, name, genre, released, backgroundImage, posterImage} = films[0];
+  const dispatch = useDispatch();
+  const [showCount, setShowCount] = useState(FILMS_COUNT_PER_STEP);
+  const filteredFilms = useAppSelector(selectFilterFilms);
+
+  useEffect(() => {
+    dispatch(getFilmsList());
+  },[dispatch]);
+
+  const getFilms = (filmss: Film[]) => filmss.slice(0, FILMS_COUNT_PER_STEP);
```

Comment:
```suggestion
  const getFilms = (films: Film[]) => films.slice(0, FILMS_COUNT_PER_STEP);
```
а зачем это функцией? Почему не просто сразу в переменную? slicedFilms на пример?